### PR TITLE
Packaging dgpsi for Conda Forge

### PR DESCRIPTION
**Pull request by mingdeyu**
Checklist
- [ X ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ X ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ X ] Source is from official source.
- [ X ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ X ] If static libraries are linked in, the license of the static library is packaged.
- [ X ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ X ] Build number is 0.
- [ X ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ X ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ X ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mingdeyu** [comment]
@conda-forge/staged-recipes, ready for review
@conda-forge-admin, please ping team

File: recipes/dgpsi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{% set name = "dgpsi" %}
+{% set version = "2.1b0" %}
```
Comment (ocefpaf) [inline recipes/dgpsi/meta.yaml:2]: This is a beta release and falls under cfep-05: https://github.com/conda-forge/cfep/blob/main/cfep-05.md

Comment (mingdeyu) [reply recipes/dgpsi/meta.yaml]: Have updated the package on PyPI to v2.1.0 and the version info in meta.yaml file has also been updated. Please review.

**Comment by mingdeyu** [comment]
@conda-forge/staged-recipes, ready for review
@conda-forge-admin, please ping team

**Comment by mingdeyu** [comment]
Just a quick note: grayskull said that the package is not noarch because of the way it is written, so I have left noarch out. Thanks.

**Comment by ocefpaf** [comment]
> @ocefpaf Just a quick note: grayskull said that the package is not noarch because of the way it is written, so I didn't include noarch in. Release version has been changed (see above) and now ready for review. Many thanks!

It may contain scripts and that prevents us from making it noarch. If that changes in the future you can try making it noarch.

File: recipes/dgpsi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+build:
+  number: 0
+  skip: true   # [py>=40 or py2k]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
```
Comment (ocefpaf) [inline recipes/dgpsi/meta.yaml:20]: Actually, it has compiled extensions, so it cannot be noarch!

**Pull request merged**